components/eventing-controller/controllers/subscriptionv2/jetstream/reconciler.go

```diff
@@ -197,81 +184,121 @@ func (r *Reconciler) enqueueReconciliationForSubscriptions(subs []eventingv1alph
+
+	// sync subscription status with k8s
+	if err := r.updateStatus(ctx, actualSubscription, desiredSubscription, logger); err != nil {
+		r.namedLogger().Errorf("updateStatus() error: %v", err)
+		return pkgerrors.MakeError(errFailedToUpdateStatus, err)
+	}
+
+	// update the subscription object in k8s
+	if !reflect.DeepEqual(actualSubscription.ObjectMeta.Finalizers, desiredSubscription.ObjectMeta.Finalizers) {
+		if err := r.Update(ctx, desiredSubscription); err != nil {
+			r.namedLogger().Errorf("updateStatus() error: %v", err)
+			return pkgerrors.MakeError(errFailedToAddFinalizer, err)
 		}
-		log.Debug("Removed finalizer from subscription")
 	}
+
 	return nil
 }
 
-// addFinalizerToSubscription appends the eventing finalizer to the subscription.
-func (r *Reconciler) addFinalizerToSubscription(sub *eventingv1alpha2.Subscription, log *zap.SugaredLogger) error {
-	sub.ObjectMeta.Finalizers = append(sub.ObjectMeta.Finalizers, eventingv1alpha2.Finalizer)
-	// to avoid a dangling subscription, we update the subscription as soon as the finalizer is added to it
-	if err := r.Client.Update(context.Background(), sub); err != nil {
-		return pkgerrors.MakeError(errFailedToAddFinalizer, err)
+// updateStatus updates the status to k8s if modified.
+func (r *Reconciler) updateStatus(ctx context.Context, oldSubscription,
+	newSubscription *eventingv1alpha2.Subscription, logger *zap.SugaredLogger) error {
+	// compare the status taking into consideration lastTransitionTime in conditions
+	if objectv2.IsSubscriptionStatusEqual(oldSubscription.Status, newSubscription.Status) {
+		return nil
 	}
-	log.Debug("Added finalizer to subscription")
+
+	// update the status for subscription in k8s
+	if err := r.Status().Update(ctx, newSubscription); err != nil {
+		return pkgerrors.MakeError(errFailedToUpdateStatus, err)
+	}
+	logger.Debugw("Updated subscription status",
+		"oldStatus", oldSubscription.Status, "newStatus", newSubscription.Status)
+
 	return nil
 }
 
-// syncInitialStatus keeps the latest cleaned EventTypes and jetStreamTypes in the subscription.
-func (r *Reconciler) syncInitialStatus(subscription *eventingv1alpha2.Subscription) bool {
-	statusChanged := false
-	cleanedTypes := jetstream.GetCleanEventTypes(subscription, r.cleaner)
-	if !reflect.DeepEqual(subscription.Status.Types, cleanedTypes) {
-		subscription.Status.Types = cleanedTypes
-		statusChanged = true
+// addFinalizer appends the eventing finalizer to the subscription.
+func (r *Reconciler) addFinalizer(ctx context.Context,
+	sub *eventingv1alpha2.Subscription, log *zap.SugaredLogger) (ctrl.Result, error) {
+	sub.ObjectMeta.Finalizers = append(sub.ObjectMeta.Finalizers, eventingv1alpha2.Finalizer)
+	log.Errorf("new Finalizers %v", sub.ObjectMeta.Finalizers)
+
+	// to avoid a dangling subscription, we update the subscription as soon as the finalizer is added to it
+	return ctrl.Result{}, r.syncSubscriptionStatus(ctx, sub, nil, log)
```

Comment:
While reading the code, I can't reason, why we need to sync subscription status if only a finalizer is being added

Reply:
updated.